llvm/lib/Analysis/DependenceAnalysis.cpp

```diff
@@ -3625,21 +3621,22 @@ DependenceInfo::depends(Instruction *Src, Instruction *Dst) {
     break; // The underlying objects alias; test accesses for dependence.
   }
 
-  // establish loop nesting levels
-  establishNestingLevels(Src, Dst);
-  LLVM_DEBUG(dbgs() << "    common nesting levels = " << CommonLevels << "\n");
-  LLVM_DEBUG(dbgs() << "    maximum nesting levels = " << MaxLevels << "\n");
-
-  FullDependence Result(Src, Dst, PossiblyLoopIndependent, CommonLevels);
-  ++TotalArrayPairs;
+  if (DstLoc.Size != SrcLoc.Size) {
```

Comment:
This is #116630?

Reply:
Correct.  This patch is on top of #116630.
I can either rebase this PR on top of nothing, or squash both patches in one.
I would prefer to separately commit #116630 first, and then work on to this PR.